fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/LocatableResolver.java

```diff
@@ -483,6 +524,95 @@ private CompletableFuture<ResolverResult> resolveWithCache(@Nonnull FDBRecordCon
         });
     }
 
+    /**
+     * Read a value within the context of a single transaction. This will use the transaction passed in when
+     * reading from the database (unlike some of the methods in this class which will just use it as the
+     * basis of child transactions). If the name has not been previously inserted into the resolver, it
+     * will return {@code null}. Additionally, the isolation properties are somewhat relaxed, in that it can
+     * read values committed in other transactions (from the {@linkplain FDBDatabase#getDirectoryCache(int)
+     * directory cache}), though it won't return any uncommitted data.
```

Comment:
> Also, I say "intention" above, because I think the resolver state cache isn't quite right, because all scoped directory layers share the same "resolver state cache", which seems wrong. (Instead, the value should be cached by scope.)

I don't think that's correct. Based on my reading, the "resolver state cache" is keyed based on the `LocatableResolver` which has the location in `equals`, so I think it's doing the right thing.